### PR TITLE
fix(test): ensure that binary is built before the test runs

### PR DESCRIPTION
**Pull request by samueltardieu**
## Description of change

The `app.rs` test requires that the `binsider` binary has been built. By moving the test into the `tests` directory, `cargo` will ensure that the binary has been built before the test can be run.

## How has this been tested? (if applicable)

By running `cargo test` from a clean directory. It fails before this patch.

Note that this does not fix another issue: the test will still fail when running `cargo test --release` as it requires the binary to be available in `target/debug/`.

### CODE REVIEW

**Review by Omega-Centauri-21 — APPROVED** [review]
These changes will improve the reliability and organization of the tests

File: tests/app.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+use binsider::{app::Analyzer, error::Result, file::FileInfo, prelude::Event};
+use std::{fs, path::PathBuf, sync::mpsc};
+
+fn get_test_path() -> PathBuf {
+    PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .join("target")
+        .join("debug")
+        .join(env!("CARGO_PKG_NAME"))
```
Comment (orhun) [inline tests/app.rs:8]: I think we can now use `CARGO_BIN_EXE_binsider` here:

> The absolute path to a binary target’s executable. This is only set when building an [integration test](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#integration-tests) or benchmark.

More info: https://doc.rust-lang.org/cargo/reference/environment-variables.html

Can you try it out? 😊

**Review by orhun — COMMENTED** [review]
Thanks for the PR! This is smart.

File: tests/app.rs (hunk 1 above)
Comment (samueltardieu) [reply tests/app.rs]: Indeed, it works fine. Pushed.

**Review by orhun — APPROVED** [review]
LGTM!

**Pull request merged**